### ISSUE TITLE
Use of `ComputeMethod.OnWrite`-feature in a bone that is referenced as refKey in `RelationalBone` is always

### ISSUE THREAD

**Issue by phorward** [viur-framework/viur-core#943]
Given the following data model, the bone `visible` in the RelationalBone `portal` is ALWAYS saved as None (null) inside the relation, as it is calculated in `BaseBone.serialize()` which is executed by `RefSkel` to obtain the values.

```python
# Referenced Skeleton
class PortalSkel(skeleton.Skeleton):
    active = BooleanBone(
        descr="Aktiv",
        defaultValue=True,
    )

    hidden = BooleanBone(
        descr="Versteckt",
        defaultValue=False,
    )

    visible = BooleanBone(
        descr="Sichtbar im Dashboard",
        compute=Compute(lambda skel: skel["active"] and not skel["hidden"], ComputeInterval(ComputeMethod.OnWrite))
    )

# Skeleton where portal.dest.visible is always None!
class UserPortalSkel(skeleton.Skeleton):
    user = UserBone(
        descr="Benutzer",
        readOnly=True,
        consistency=RelationalConsistency.CascadeDeletion,
    )

    portal = RelationalBone(
        descr="Portal",
        kind="portal",
        readOnly=True,
        format="$(dest.identifier)",
        refKeys=[
            "key",
            "name",
            "visible",  # <-- DOESN'T WORK WHEN COMPUTABLE!
        ],
        consistency=RelationalConsistency.CascadeDeletion,
    )
```

The bone is broken, because "active" and "hidden" are NOT in the RefSkel that is constructed from the refKeys setting. Maybe the best solution would be to move the `ComputeMethod.OnWrite`-stuff really to `Skeleton.toDB()` where the datastore entity is actually written (which is said by the method!).

**Comment by sveneberth** [comment]
I just ran into the same error. As a workaround, it also works to include the source bones as refkeys. But of course that bloats the whole thing ...

Would be nice if we could bring #950 forward as a quick fix. We could think about a more efficient method at a later date.

**Comment by phorward** [comment]
Fixed by #950
